### PR TITLE
[14.0] shopfloor: support barcode that do not send end of line + trim value

### PR DESCRIPTION
**Pull request by jbaudoux**
Fwd port of https://github.com/OCA/wms/pull/287

### CODE REVIEW

**Review by JuMiSanAr — APPROVED** [review]

File: shopfloor_mobile_base/static/wms/src/components/searchbar/searchbar.js [hunk 1]
```diff
@@ -33,18 +36,76 @@ Vue.component("searchbar", {
...
+    },
     methods: {
         search: function (e) {
-            e.preventDefault();
             // Talk to parent
+            if (!this.entered) return;
             this.$emit("found", {
                 text: this.entered,
-                type: e.target.dataset.type,
+                type: String,
```
Comment (simahawk) [inline shopfloor_mobile_base/static/wms/src/components/searchbar/searchbar.js]: same remark as the PR on 13

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopfloor_mobile_base/static/wms/src/components/searchbar/searchbar.js (hunk 1 above)
Comment (jbaudoux) [reply shopfloor_mobile_base/static/wms/src/components/searchbar/searchbar.js]: solved

**Comment by simahawk** [comment]
waiting for approval in v13

**Comment by simahawk** [comment]
@jbaudoux pls rebase

**Comment by jbaudoux** [comment]
> @jbaudoux pls rebase

@simahawk done

**Review by simahawk — APPROVED** [review]

**Pull request merged**